server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -701,7 +722,51 @@ public void onResponse(ShardSearchRequest orig) {
                     }
                 }
                 // fork the execution in the search thread pool
-                runAsync(getExecutor(shard), () -> executeQueryPhase(orig, task, keepStatesInContext), listener);
+                runAsync(getExecutor(executorName, shard), () -> executeQueryPhase(orig, task, keepStatesInContext), listener);
+            }
+
+            @Override
+            public void onFailure(Exception exc) {
+                listener.onFailure(exc);
+            }
+        });
+    }
+
+    public void executeQueryPhaseStream(
```

Comment:
do we really need `executeQueryPhaseStream()` and its other variant in this class? 
If I'm not missing anything, all we are different here is is setting StreamSearchChannelListener to the context, which we can do in StreamTransportSearchService right?